lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControlModule.java

```diff
@@ -0,0 +1,141 @@
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import com.sun.net.httpserver.HttpServer;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.units.Duration;
+import io.trino.spi.QueryId;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.spi.security.SystemSecurityContext;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestFileBasedSystemAccessControlModule
+{
+    private static final Identity alice = Identity.forUser("alice").withGroups(ImmutableSet.of("staff")).build();
+    private static final CatalogSchemaTableName aliceView = new CatalogSchemaTableName("alice-catalog", "schema", "view");
+    private static final Optional<QueryId> queryId = Optional.empty();
+
+    private static final SystemSecurityContext ALICE = new SystemSecurityContext(alice, queryId);
+
+    @Test
+    public void noHttpClientConfigForLocalFile()
+    {
+        Injector injector = initModule(ImmutableMap.of(
+                "security.config-file", getResourcePath("file-based-system-catalog.json")));
+        SystemAccessControl accessControl = injector.getInstance(SystemAccessControl.class);
+        accessControl.checkCanCreateView(ALICE, aliceView);
+        Supplier<FileBasedSystemAccessControlRules> provider = injector.getInstance(
+                Key.get(new TypeLiteral<Supplier<FileBasedSystemAccessControlRules>>() {}));
+        assertTrue(provider instanceof LocalFileBasedSystemAccessControlRulesProvider);
+        assertThatThrownBy(() -> injector.getInstance(Key.get(HttpClient.class, ForAccessControlRules.class)))
+                .isInstanceOf(ConfigurationException.class).hasMessageContaining("is not explicitly bound");
+        assertThatThrownBy(() -> injector.getInstance(Key.get(HttpClientConfig.class)))
+                .isInstanceOf(ConfigurationException.class).hasMessageContaining("is not explicitly bound");
+    }
+
+    @Test
+    public void httpClientConfigForRestFile()
+            throws IOException
+    {
+        byte[] response = Files.readAllBytes(Path.of(
+            getResourcePath("file-based-system-catalog.json")));
+
+        try (TestingHttpServer testingHttpServer = new TestingHttpServer(2002, response)) {
+            Injector injector = initModule(ImmutableMap.of(
```

Comment:
I don't like such plumbing. The best would be to pass the configuration to query runner and then run a regular query to see if access is granted or revoked.

Reply:
Could you point me to an example about how this can be achieved? Which implementation of `QueryRunner` should I use?

Reply:
We can use a `LocalQueryRunner`

Reply:
It would be nice if we have a test via a `LocalQueryRunner`.